improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -794,25 +792,24 @@ def _mean_and_variance_to_percentiles(
 
         # Reshape forecast_at_percentiles, so the percentiles dimension is
         # first, and any other dimension coordinates follow.
-        result = (
-            restore_non_probabilistic_dimensions(
-                result, calibrated_forecast_predictor,
-                "realization", len(percentiles)))
+        result = result.reshape(
+            (len(percentiles),) + calibrated_forecast_predictor.data.shape)
+
+        for prob_coord_name in ["realization", "percentile"]:
+            if template_cube.coords(prob_coord_name, dim_coords=True):
+                prob_coord = template_cube.coord(prob_coord_name)
+                template_slice = next(template_cube.slices_over(prob_coord))
+                template_slice.remove_coord(prob_coord)
 
-        for template_cube in calibrated_forecast_predictor.slices_over(
-                "realization"):
-            template_cube.remove_coord("realization")
-            break
         percentile_cube = create_cube_with_percentiles(
-            percentiles, template_cube, result)
-        # Remove cell methods aimed at removing cell methods associated with
-        # finding the ensemble mean, which are no longer relevant.
+            percentiles, template_slice, result)
+        # Remove cell methods associated with finding the ensemble mean
         percentile_cube.cell_methods = {}
```

Comment:
@neilCrosswaite can you make sure this is fixed in your cell method PR. I don't think this line will be needed after your change.